### PR TITLE
test: fake patients now are between 99 and 0 years old

### LINKED ISSUE #38: `dateOfBirth` of `fake("Patient")` is always too recent

### PR DESCRIPTION
**Pull request by ngouy**
## What is this about

The patient faker was making use of the `recentDate` helper, which was why patients always had a very close DOB
Instead we can leverage the `randPastDate` helper
## Issue ticket numbers

FIX #38 

## How can this be tested?

They have not yet. Not sure how I can do that.
I've tested this helper individually and it works great.
https://ngneat.github.io/falso/docs/date/#randpastdate

## Known limitations/edge cases

_none_

### CODE REVIEW

File: packages/test-support/r4b/fakes.ts [hunk 1]
```diff
@@ -8374,7 +8375,7 @@ function fakePatient(): Patient {
     name: [fakeHumanName()],
     telecom: [fakeContactPoint()],
     gender: rand(["male", "female", "other", "unknown"]),
-    birthDate: randRecentDate().toISOString().slice(0, 10),
+    birthDate: randPastDate({ years: 99 }).toISOString().slice(0, 10),
```
Comment (julienblin) [inline packages/test-support/r4b/fakes.ts:8378]: This is the correct final fix - _but_, this file is code-generated, driven by a pretty wonky code.
So this is not the place to do the update.

The template to generate this file is here: https://github.com/bonfhir/bonfhir/blob/main/packages/test-support/r4b/fakes.ts.hbs

However, to drive this customization, you need to have look here: https://github.com/bonfhir/bonfhir/blob/main/packages/test-support/template-helpers.js
and specify a different randomizer for `Patient.birthDate`.

Then, you can rerun the code generation by using `yarn codegen` at the root of the `test-support` package.
Let me know if you need a bit more explanation :-)

**Review by julienblin — CHANGES_REQUESTED** [review]

File: packages/test-support/r4b/fakes.ts (hunk 1 above)
Comment (ngouy) [reply packages/test-support/r4b/fakes.ts:8378]: interesting. I wasn't sure if auto-generated or not. No indication at the top of the file whatsoever

I will take a look at your comment

Comment (julienblin) [reply packages/test-support/r4b/fakes.ts:8378]: Interesting point. Are you aware of a convention for such headers in Typescript / JavaScript?

**Review by julienblin — DISMISSED** [review]
Love it! Thanks!

**Review by julienblin — CHANGES_REQUESTED** [review]
The code is great - thank you!

One last thing: let's work on the commit line as it should include the `test-support` scope to be picked up by the CHANGELOG generator; we do use [Conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) on this project.

**Comment by ngouy** [comment]
>One last thing: let's work on the commit line as it should include the test-support scope to be picked up by the CHANGELOG generator; we do use [Conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) on this project

That part I don't get. I thought the `test: <description>` was enough?
When I look at the commit list it's hard to see consistency (for example: the last one is "just" `Feat-ui-components-package`)

Some have scope some don't have. Is there a rule on when we put one / do we have a list?

So here, (just to make sure), we want: `test(support): fake patients now are between 99 and 0 years old` ?

**Comment by julienblin** [comment]
> > One last thing: let's work on the commit line as it should include the test-support scope to be picked up by the CHANGELOG generator; we do use [Conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) on this project
> 
> That part I don't get. I thought the `test: <description>` was enough? When I look at the commit list it's hard to see consistency (for example: the last one is "just" `Feat-ui-components-package`)
> 
> Some have scope some don't have. Is there a rule on when we put one / do we have a list?
> 
> So here, (just to make sure), we want: `test(support): fake patients now are between 99 and 0 years old` ?

2 things:
 - we do not have conventions on merge commits, because those are not used to generate changelogs - thus the "Feat-ui-components-package"; only **non-merge commits** are used
 - sometimes I do bypass the scope if I don't want a commit description to end up in the changelog, as easy as this
 
In this case, I feel it should be as it's a clear change in behavior. So here is what I suggest:

`fix(test-support): birthDate attributes are now between 0 and 99 years old`

**Comment by ngouy** [comment]
interesting. Thanks for the details and the example @julienblin  it helps a lot

**Review by julienblin — APPROVED** [review]
🥳

**Comment by julienblin** [comment]
I merged the PR as I want to release some packages today.
